Cubical/Algebra/Field/Base.agda

```diff
@@ -145,17 +152,38 @@ _$_ : {R S : Field ℓ} → (φ : FieldHom R S) → (x : ⟨ R ⟩) → ⟨ S 
 FieldEquiv→FieldHom : {A B : Field ℓ} → FieldEquiv A B → FieldHom A B
 FieldEquiv→FieldHom (e , eIsHom) = e .fst , eIsHom
 
-isPropIsField : {R : Type ℓ} (0r 1r : R) (_+_ _·_ : R → R → R) (-_ : R → R) (_[_]⁻¹ : (x : R) → ¬ (x ≡ 0r) → R)
-             → isProp (IsField 0r 1r _+_ _·_ -_ _[_]⁻¹)
-isPropIsField 0r 1r _+_ _·_ -_ _[_]⁻¹ (isfield RR RC RD) (isfield SR SC SD) =
+isPropIsField : {R : Type ℓ} (0r 1r : R) (_+_ _·_ : R → R → R) (-_ : R → R)
+             → isProp (IsField 0r 1r _+_ _·_ -_)
+isPropIsField {R = R} 0r 1r _+_ _·_ -_ H@(isfield RR RC RD) (isfield SR SC SD) =
   λ i → isfield (isPropIsCommRing _ _ _ _ _ RR SR i)
-                   (isPropInv RC SC i) (isProp→⊥ RD SD i)
+                   (isPropInv RC SC i) (isProp¬ _ RD SD i)
   where
   isSetR : isSet _
   isSetR =  RR .IsCommRing.·IsMonoid .IsMonoid.isSemigroup .IsSemigroup.is-set
 
-  isPropInv : isProp ((x : _) → (≢0 : ¬ (x ≡ 0r)) → x · (x [ ≢0 ]⁻¹) ≡ 1r)
-  isPropInv = isPropΠ2 λ _ _ → isSetR _ _
+  isPropInv : isProp ((x : _) → ¬ x ≡ 0r → Σ[ y ∈ R ] x · y ≡ 1r)
+  isPropInv = isPropΠ2 (λ x _ → Units.inverseUniqueness (Field→CommRing (_ , fieldstr _ _ _ _ _ H)) x)
+
+𝒮ᴰ-Field : DUARel (𝒮-Univ ℓ) FieldStr ℓ
+𝒮ᴰ-Field =
+  𝒮ᴰ-Record (𝒮-Univ _) IsFieldEquiv
+    (fields:
+      data[ 0r ∣ null ∣ pres0 ]
+      data[ 1r ∣ null ∣ pres1 ]
+      data[ _+_ ∣ bin ∣ pres+ ]
+      data[ _·_ ∣ bin ∣ pres· ]
+      data[ -_ ∣ autoDUARel _ _ ∣ pres- ]
+      prop[ isField ∣ (λ _ _ → isPropIsField _ _ _ _ _) ])
+  where
+  open FieldStr
+  open IsFieldHom
+
+  -- faster with some sharing
+  null = autoDUARel (𝒮-Univ _) (λ A → A)
+  bin = autoDUARel (𝒮-Univ _) (λ A → A → A → A)
+
+FieldPath : (R S : Field ℓ) → FieldEquiv R S ≃ (R ≡ S)
+FieldPath = ∫ 𝒮ᴰ-Field .UARel.ua
 
-  isProp→⊥ : ∀ {A : Type ℓ} → isProp (A → ⊥)
-  isProp→⊥ = isPropΠ λ _ → isProp⊥
+uaFieldRing : {A B : Field ℓ} → FieldEquiv A B → A ≡ B
```

Comment:
Why does this one have 'Ring' in its name?

Reply:
Eh... Embarrassing...